File: frontend/lib/models/task_list.dart

enum TaskListCategory {
  template,
  toDoList,
}

class TaskList {
  final int id;
  final String title;
  final TaskListCategory category;
  final List<int> taskIds;
  final bool archived;

  TaskList({
    required this.id,
    required this.title,
    required this.category,
    this.taskIds = const [],
    this.archived = false,
  });
}

class TaskListMetadata {
  final int id;
  final int total;
  final int completed;

  TaskListMetadata({
    required this.id,
    required this.total,
    required this.completed,
  });
}

File: android/app/src/main/java/com/tomyedwab/yellowstone/models/TaskList.kt
package com.tomyedwab.yellowstone.models

import com.google.gson.annotations.SerializedName

data class TaskList(
        @SerializedName("Id") val id: Int,
        @SerializedName("Title") val title: String,
        @SerializedName("Category") val category: String,
        @SerializedName("Archived") val archived: Boolean
)

data class TaskListResponse(@SerializedName("TaskLists") val taskLists: List<TaskList>)

data class TaskMetadata(
        @SerializedName("ListId") val listId: Int,
        @SerializedName("Total") val totalTasks: Int,
        @SerializedName("Completed") val completedTasks: Int
)

data class TaskMetadataResponse(@SerializedName("Metadata") val metadata: List<TaskMetadata>)
